host/CMakeLists.txt

```diff
@@ -338,8 +338,7 @@ if (USE_LIBSGX)
       IMPORTED_LOCATION ${WINSYSLOCATION}
       IMPORTED_IMPLIB ${LIBSGX_QE})
   endif ()
-
-  target_link_libraries(oehost PUBLIC sgx_enclave_common sgx_dcap_ql)
+  target_link_libraries(oehost PUBLIC $<BUILD_INTERFACE:sgx_dcap_ql> $<BUILD_INTERFACE:sgx_enclave_common>)
```

Comment:
When building the SDK, the two sgx libraries will be considered target dependencies by this link line.
When consuming the SDK (using the cmake configuration export), these two are detected by the openenclave-config.cmake package configuration file.

You'll find a line in openenclave-config.cmake.in with this:
```
target_link_libraries(openenclave::oehost INTERFACE openenclave::sgx_enclave_common openenclave::sgx_dcap_ql)
```

Where the openenclave::sgx_enclave_common & openenclave::sgx_dcap_ql libraries are found using find_library first.

This line solves the main problem I had with this entire PR, trying to get oehost to not reference "sgx_enclave_common" in link lines. Previously because these are `PUBLIC` dependencies of oehost for the cmake configuration export, when linking an executable in the samples with openenclave::oehost, it automatically tried to link against sgx_enclave_common and sgx_dcap_ql, not the namespaced openenclave::sgx_enclave_common and openenclave::sgx_dcap_ql. We can't namespace sgx_enclave_common and sgx_dcap_ql (using `install(TARGETS sgx_enclave_common ...)` )because they are `IMPORTED` libraries, and that isn't legal to do.

I learned way more about cmake than I wanted to know trying to figure out the best way to do this with @andschwa 's guidance.